File: native_client/tests/gdb/kill.py
# -*- python -*-
# Copyright (c) 2014 The Native Client Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

import gdb_test


class KillTest(gdb_test.GdbTest):

  def test_kill(self):
    # Test that you can stop on a breakpoint, then kill the program being
    # debugged.
    self.gdb.Command('break test_kill')
    self.gdb.ResumeAndExpectStop('continue', 'breakpoint-hit')
    self.gdb.Kill()

  def tearDown(self):
    # Test program should end first with the kill return code.
    # Intentionally bypass superclass's tearDown as it assumes gdb exits first.
    self.AssertSelLdrExits()
    self.gdb.Quit()
    self.gdb.Wait()


if __name__ == '__main__':
  gdb_test.Main()

File: native_client/src/untrusted/nacl/stubs/kill.c
/*
 * Copyright 2010 The Native Client Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

/*
 * Stub routine for `kill' for porting support.
 */

#include <sys/types.h>
#include <signal.h>
#include <errno.h>

int kill(int pid, int sig) {
  errno = ENOSYS;
  return -1;
}
